src/aioswitcher/device/tools.py

```diff
@@ -165,6 +166,34 @@ def convert_str_to_devicetype(device_type: str) -> DeviceType:
     return DeviceType.MINI
 
 
+def convert_token_to_packet(
+    device_type: DeviceType, token: Union[str, None] = None
+) -> str:
+    """Return token packet from token if is valid, otherwise empty string."""
+    is_token_needed = bool(device_type.token_needed)
+    is_token_not_empty = token is not None and str(token) != ""
```

Comment:
It is done.

Reply:
That's better. Thanks.

We now have:
```python
is_token_needed = device_type.token_needed
is_token_not_empty = bool(token)
if is_token_needed:
    if is_token_not_empty and token is not None:
```

Why do we need a variable named `is_token_needed` to store the value of `device_type.token_needed`, mainly if it is only evaluated once in the next two rows?

Also, Python's built-in _bool_ function already verifies the token `is not None`. (See [here](https://docs.python.org/3/library/functions.html#bool) and [here](https://www.w3schools.com/python/ref_func_bool.asp)).

You can change the above code to the following, and it will have the same effect:
```python
if device_type.token_needed:
    if bool(token):
```

Or, if you insist:
```python
is_token_not_empty = bool(token)
if device_type.token_needed:
    if is_token_not_empty:
```

---

But honestly, if you're trying to make this more readable, I would suggest reversing the `if` branches (please forgive any bugs, this was written in-comment):
```python
if not device_type.token_needed:
    return None

if not bool(token):
    raise RuntimeError("a token is needed but missing")

try:
    encrypted_value = b64decode(bytes(token, "utf-8"))
    cipher = AES.new(b"jzNrAOjc%lpg3pVr5cF!5Le06ZgOdWuJ", AES.MODE_ECB)
    decrypted_value = cipher.decrypt(encrypted_value)
    unpadded_decrypted_value = unpad(decrypted_value, AES.block_size)
    return hexlify(unpadded_decrypted_value).decode()
except (KeyError, ValueError) as ve:
    raise RuntimeError("convert token to packet was not successful") from ve
```

Reply:
> Why do we need a variable named is_token_needed to store the value of device_type.token_needed, mainly if it is only evaluated once in the next two rows?

I just thought that the naming made more sense and was readable but I can also not use the variables..

> Also, Python's built-in bool function already verifies the token is not None. (See [here](https://docs.python.org/3/library/functions.html#bool) and [here](https://www.w3schools.com/python/ref_func_bool.asp)).

I tried that.. but I was getting a fail by linting.. I had to add `is not None` directly so I could pass it.
That is also true for those:

> if device_type.token_needed:
>     if bool(token): 

and

> is_token_not_empty = bool(token)
    if device_type.token_needed:
       if is_token_not_empty:

As for that:

> But honestly, if you're trying to make this more readable, I would suggest reversing the if branches (please forgive any bugs, this was written in-comment):

I can try that..

Reply:
What was the linting error?

Reply:
> But honestly, if you're trying to make this more readable, I would suggest reversing the if branches (please forgive any bugs, this was written in-comment):

I implemented the same right now, you will see in the status check the lining issue I am facing..

Reply:
By the way, I really dislike return statements occurring out of context. This just screams "error-prone." Please take another look at my example [here](https://github.com/TomerFi/aioswitcher/pull/766#discussion_r1567465804).

```python
# change this
    ...
        token_packet = hexlify(unpadded_decrypted_value).decode()
    except (KeyError, ValueError) as ve:
        raise RuntimeError("convert token to packet was not successful") from ve
    return token_packet

# to this
    ...
        return hexlify(unpadded_decrypted_value).decode()
    except (KeyError, ValueError) as ve:
        raise RuntimeError("convert token to packet was not successful") from ve
```

Reply:
Done.

Reply:
@TomerFi - can you review and approve merging please 🥺